pkg/controller/actuator_reconcile.go

```diff
@@ -110,13 +111,24 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, network *extens
 
 	if cluster.Shoot.Spec.Networking != nil && cluster.Shoot.Spec.Networking.Nodes != nil && len(*cluster.Shoot.Spec.Networking.Nodes) > 0 {
 		autodetectionMode := fmt.Sprintf("cidr=%s", *cluster.Shoot.Spec.Networking.Nodes)
-
+		//for dualstack we should get shoot.Status.Networking.NodeCIDRs
 		if ipFamilies.Has(extensionsv1alpha1.IPFamilyIPv4) {
 			networkConfig.IPv4.AutoDetectionMethod = &autodetectionMode
 		}
-
+		autodetectionModeV6 := autodetectionMode
+		if cluster.Shoot.Status.Networking != nil && cluster.Shoot.Status.Networking.Nodes != nil && len(cluster.Shoot.Status.Networking.Nodes) > 0 {
+			for i, nodeCidr := range cluster.Shoot.Status.Networking.Nodes {
+				_, cidr, err := net.ParseCIDR(nodeCidr)
+				if err != nil {
+					return err
+				}
+				if cidr.IP.To4() == nil {
+					autodetectionModeV6 = fmt.Sprintf("cidr=%s", cluster.Shoot.Status.Networking.Nodes[i])
+				}
+			}
+		}
```

Comment:
There might be several ranges per IP family, e.g. one range per availability zone. Please check if we can configure all of them. [Documentation](https://docs.tigera.io/calico/latest/networking/ipam/ip-autodetection) seems to indicate that it is possible.